x-pack/legacy/plugins/maps/public/layers/styles/vector/properties/dynamic_color_property.js

```diff
@@ -232,9 +236,40 @@ export class DynamicColorProperty extends DynamicStyleProperty {
   }
 
   _getColorRampStops() {
-    return this._options.useCustomColorRamp && this._options.customColorRamp
-      ? this._options.customColorRamp
-      : [];
+    if (this._options.useCustomColorRamp && this._options.customColorRamp) {
+      return this._options.customColorRamp;
+    }
+
+    if (!this._options.color) {
+      return [];
+    }
+
+    const rangeFieldMeta = this.getRangeFieldMeta();
+    if (!rangeFieldMeta) {
+      return [];
+    }
+
+    const numberOfLegendColors = 5;
```

Comment:
Maybe we should move this to the top of the file and make it something like: `const NUM_LEGEND_COLORS = 5;` We might also want a comment saying where 5 comes from.